presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -404,6 +405,15 @@ private static TableStatistics getTableStatistics(
 
         TableStatistics.Builder result = TableStatistics.builder();
         result.setRowCount(Estimate.of(rowCount));
+
+        OptionalDouble optionalAverageSizePerPartition = calculateAverageSizePerPartition(statistics.values());
+        if (optionalAverageSizePerPartition.isPresent()) {
+            double averageSizePerPartition = optionalAverageSizePerPartition.getAsDouble();
+            verify(averageSizePerPartition >= 0, "averageSizePerPartition must be greater than or equal to zero");
```

Comment:
nit: include the actual value into the error message in case it ever fails

e.g.: `verify(averageSizePerPartition >= 0, "averageSizePerPartition must be greater than or equal to zero: %s", averageSizePerPartition)`